modules/pricing/pricing-repository.ts

```diff
@@ -0,0 +1,185 @@
+                            tokenAddress: item.address,
+                            timestamp: hourlyTimestamp,
+                            chain: item.chain,
+                        },
+                    },
+                    update: {
+                        price: item.price,
+                        close: item.price,
+                        updatedBy: item.updatedBy,
+                        updatedAt: item.updatedAt,
+                    },
+                    create: {
+                        tokenAddress: item.address,
+                        chain: item.chain,
+                        timestamp: hourlyTimestamp,
+                        price: item.price,
+                        high: item.price,
+                        low: item.price,
+                        open: item.price,
+                        close: item.price,
+                        updatedBy: item.updatedBy,
+                        updatedAt: item.updatedAt,
+                    },
+                }),
+            );
+
+            // Update or create current price in TokenCurrentPrice table
+            operations.push(
+                prisma.prismaTokenCurrentPrice.upsert({
+                    where: {
+                        tokenAddress_chain: {
+                            tokenAddress: item.address,
+                            chain: item.chain,
+                        },
+                    },
+                    update: {
+                        price: item.price,
+                        timestamp: hourlyTimestamp,
+                        updatedBy: item.updatedBy,
+                        updatedAt: item.updatedAt,
+                    },
+                    create: {
+                        tokenAddress: item.address,
+                        chain: item.chain,
+                        timestamp: hourlyTimestamp,
+                        price: item.price,
+                        updatedBy: item.updatedBy,
+                        updatedAt: item.updatedAt,
+                    },
+                }),
+            );
+        }
+
+        await prismaBulkExecuteOperations(operations);
+
+        return priceItems.map((item) => item.address);
+    }
+
+    private collectAllTokenAddresses(
+        tokens: { address: string; underlyingTokenAddress?: string | null }[],
+        swaps: SwapEvent[],
+    ): string[] {
+        const addresses = new Set<string>();
+
+        // Add all target token addresses
+        tokens.forEach((token) => {
+            addresses.add(token.address);
+            // Add underlying token addresses for ERC4626/Aave tokens
+            if (token.underlyingTokenAddress) {
+                addresses.add(token.underlyingTokenAddress);
+            }
+        });
+
+        // Add all swap token addresses (for other tokens in swaps)
+        swaps.forEach((swap) => {
+            addresses.add(swap.payload.tokenIn.address);
+            addresses.add(swap.payload.tokenOut.address);
+        });
```

Comment:
dont think this is needed. In `getTokensForPricing` you loop through `tokens` to produce the end result and get prices for them, so adding more here wont make a difference if their not in the original `tokens` list anyway. Makes sense for the underlying prices but dont think you need the swap addresses

Reply:
i added that just in case there is no token in the database, but then we don't price them anyways, so i'm cleaning it up